ecommerce/api_test.py

```diff
@@ -719,3 +734,92 @@ def test_not_redeemable(self):
+        """
+        If a user is not enrolled a 404 should be raised when getting the price
+        """
+        course_run, user = create_purchasable_course_run()
+        ProgramEnrollment.objects.filter(program=course_run.course.program, user=user).delete()
+        try:
+            calculate_run_price(course_run, user)
+        except Http404:
+            pass
+
+    def test_calculate_run_price_coupon(self):
+        """
+        If there is a coupon calculate_run_price should use calculate_coupon_price to get the discounted price
+        """
+        course_run, user = create_purchasable_course_run()
+        coupon = CouponFactory.create(content_object=course_run)
+        UserCoupon.objects.create(coupon=coupon, user=user)
+        discounted_price = 5
+        with patch('ecommerce.api.calculate_coupon_price', autospec=True) as _calculate_coupon_price:
+            _calculate_coupon_price.return_value = discounted_price
+            assert calculate_run_price(course_run, user) == (discounted_price, coupon)
+        fa_price = get_formatted_course_price(course_run.course.program.programenrollment_set.first())['price']
+        _calculate_coupon_price.assert_called_with(coupon, fa_price, course_run.edx_course_key)
+
+    def test_percent_discount(self):
+        """
+        Assert the price with a percent discount
+        """
+        course_run, _ = create_purchasable_course_run()
+        price = Decimal(5)
+        coupon = CouponFactory.create(
+            content_object=course_run, amount_type=Coupon.PERCENT_DISCOUNT, amount=Decimal("0.3")
+        )
+        assert calculate_coupon_price(coupon, price, course_run.edx_course_key) == price * (1 - coupon.amount)
+
+    def test_fixed_discount(self):
+        """
+        Assert the price with a fixed discount
+        """
+        course_run, _ = create_purchasable_course_run()
+        price = Decimal(5)
+        coupon = CouponFactory.create(
+            content_object=course_run, amount_type=Coupon.FIXED_DISCOUNT, amount=Decimal("1.5")
+        )
+        assert calculate_coupon_price(coupon, price, course_run.edx_course_key) == price - coupon.amount
+
+    def test_calculate_coupon_price(self):
+        """
+        Assert that the price is not adjusted if the amount type is unknown
+        """
+        course_run, _ = create_purchasable_course_run()
+        price = Decimal('0.3')
+        coupon = CouponFactory.create(content_object=course_run)
+        # Use manager to get around validation
+        Coupon.objects.filter(id=coupon.id).update(amount_type='xyz')
+        coupon.refresh_from_db()
+        assert calculate_coupon_price(coupon, price, course_run.edx_course_key) == price
+
+    def test_coupon_allowed(self):
+        """
+        Assert that the price is not adjusted if the coupon doesn't affect the course run
+        """
+        course_run, _ = create_purchasable_course_run()
+        price = Decimal('0.3')
+        coupon = CouponFactory.create()
```

Comment:
Can you be a little more explicit, here? Why doesn't the coupon affect the course run? It could be as simple as adding a line like this:
```python
assert coupon.content_object != course_run
```